packages/app/src/pages/SwapPage/index.tsx

```diff
@@ -48,15 +55,29 @@ const getValidationText = (
 const getValidationState = ({
   swapState,
   previewAmount,
+  slippage,
+  balances,
   hasLiquidity,
 }: StateParams): ValidationStateEnum => {
-  if (!swapState?.to || !swapState?.from) {
+  if (!swapState?.coinFrom || !swapState?.coinTo) {
     return ValidationStateEnum.SelectToken;
   }
   if (!swapState?.amount) {
     return ValidationStateEnum.EnterAmount;
   }
-  if (!swapState.hasBalance) {
+  if (
+    !swapState.hasBalance ||
+    (swapState.direction === ActiveInput.to &&
+      calculatePriceWithSlippage(
```

Comment:
couldn't that be in a variable? it's a bit not readable having all these things inside the condition